### PR TITLE
added: symmetrized version of beta refinement

### PR DESCRIPTION
**Pull request by akva2**

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this with downstreams ifem-poisson=31 please

File: src/SIM/AdaptiveSIM.C [hunk 1]
```diff
@@ -634,3 +634,18 @@ bool AdaptiveSIM::writeGlv (const char* infile, int iStep)
...
+  std::stringstream str;
+  str << "errors_" << step-1 << ".txt";
+  std::ofstream of(str.str());
+  of.precision(16);
+  of << "# Index   Error" << std::endl;
+  for (size_t i = 0; i < refineSize; ++i)
+    of << errors[i].second << " " << errors[i].first << std::endl;
+  of << "\n# ------------- Below here not refined ---------#\n\n";
+  for (size_t i = refineSize; i < errors.size(); ++i)
+    of << errors[i].second << " " << errors[i].first << std::endl;
```
Comment (kmokstad) [inline src/SIM/AdaptiveSIM.C]: OK, but alternative syntax for 646-650:
```
size_t i = 0;
for (const DblIdx& error : errors) {
  of << error.second <<" "<< error.first << std::endl;
  if (++i == refineSize)
    of <<"\n# ------------- Below here not refined ---------#\n\n";
}
```

File: src/SIM/AdaptiveSIM.C [hunk 2]
```diff
@@ -472,13 +479,21 @@ bool AdaptiveSIM::adaptMesh (int iStep, std::streamsize outPrec)
...
+  if (threshold == NONE || threshold == SYMMETRIZED)
     refineSize = ceil(errors.size()*beta/100.0);
   else
     refineSize = std::upper_bound(errors.begin(), errors.end(), DblIdx(limit,0),
                                   std::greater_equal<DblIdx>())
                - errors.begin();
 
+  if (threshold == SYMMETRIZED) {
+    double fErr = errors[refineSize-1].first;
+    double fEps = std::abs(fErr)*symmEps;
```
[the remark this answers is not in the document]
Comment (akva2) [reply src/SIM/AdaptiveSIM.C]: nothing, changed.

File: src/SIM/AdaptiveSIM.C (hunk 1 above)
Comment (akva2) [reply src/SIM/AdaptiveSIM.C]: changed.

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Comment by akva2** [comment]
jenkins build this with downstreams ifem-poisson=31 please

**Pull request merged**